docs/recap/bitvm2_fast_recap.tex

```diff
@@ -0,0 +1,229 @@
+while providing a fail-safe against incorrect or malicious
+operations. This document provides a concise overview of BitVM2,
+highlighting its potential to unlock new possibilities for secure and
+scalable program execution on Bitcoin.
+
+\section{Program splitting}\label{sec:program-splitting}
+
+\begin{quote}
+  Let there be a large program \(f\), which can be described in Bitcoin
+  Script. We want to compute it \textbf{\textbf{on-chain}}, i.e., find
+  such an \(f(x) = y\). To achieve this, we divide the program into
+  \(n\) sub-programs \(f_1, \ldots, f_n\) and their corresponding
+  intermediate states \(z_1, \ldots, z_n\), such that:
+
+  \begin{equation}
+    \begin{aligned}
+      f_1(x) &= z_1 \\
+      f_2(z_1) &= z_2 \\
+      &\cdots \\
+      f_n(z_{n-1}) &= y
+    \end{aligned}
+  \end{equation}
+\end{quote}
+
+However, the user (referred in BitVM2 as the operator) only needs to
+prove that the given program \(f\) indeed returns \(y\) for \(x\), or
+\textbf{give others the opportunity} to \textbf{disprove this
+fact}. In our case, this means giving challengers the ability to
+prove that at least one of the sub-programs \(f_i(z_{i-1}) \neq z_i\).
+
+\section{\textbf{Assert} transaction}\label{sec:assert-tx}
+
+To achieve this, the user publishes an \textbf{Assert} transaction,
+which has one output with multiple possible spending scenarios:
+
+\begin{enumerate}
+  \item \textbf{PayoutScript} (\textbf{LockTime} + signature) --- the
+    transaction has passed verification, and the operator can spend the
+    output, thereby confirming the statement \(f(x) = y\).
+  \item \textbf{DisproveScript} --- one of the challengers has found a
+    discrepancy in the intermediate states \(z_i\), \(z_{i-1}\) and the
+    sub-program \(f_i\). In other words, they have proven that
+    \(f_i(z_{i-1}) \neq z_i\), and thus, they can spend the output.
+\end{enumerate}
+
+\subsection{\texttt{DispoveScript}}\label{sec:dispove-script}
+
+\texttt{DispoveScript} is part of the MAST tree in a Taproot address,
+which allows challengers to claim the transaction amount for states
+\(z_i\), \(z_{i-1}\), and sub-program \(f_i\), is called
+\(DisproveScript_i\) in BitVM2:
+
+\begin{verbatim}
+// push z_i, z_{i-1} onto the stack
+{ z_i     }
+{ z_(i-1) }
+// compute f(z_{i-1})
+{ f_i     }
+// ensure that f_i(z_{i-1}) != z_i
+OP_EQUAL
+OP_NOT
+\end{verbatim}
+
+In fact, this script does not need a \texttt{script\_sig}, as with the
+correct \(z_i\) and \(z_{i-1}\), it will always execute
+successfully. Therefore, to restrict spending capability, a Lamport
+signature verification is added to the script.
+
+\subsection{Lamport signature}\label{sec:lamport-signature}
+
+Unlike other digital signature algorithms, the Lamport signature uses
+a pair of random secret and public keys \((sk_{\mathcal{M}},
+pk_{\mathcal{M}})\) that can sign and verify only any message from the
+space \(\mathcal{M} = {\{0, 1\}}^{l}\) of \(l\) -bit messages.
```

Comment:
I prefer using `\ell`, looks a bit more elegantly

Reply:
Also, `\mathsf{sk}` instead of pure `sk`

Reply:
> I prefer using \ell, looks a bit more elegantly

Oh, great, I was looking for it

> Also, \mathsf{sk} instead of pure sk

The same for $\mathsf{pk}$ ?

Reply:
Yes, the same goes for public key notation